core/src/main/java/org/ldaptive/pool/IdlePruneStrategy.java

```diff
@@ -81,8 +82,11 @@ public long getPrunePeriod()
    *
    * @param  period  to set
    */
-  public void setPrunePeriod(final long period)
+  public void setPrunePeriod(final Duration period)
   {
+    if (period == null || period.isNegative()) {
+      throw new IllegalArgumentException("Prune period cannot be null or negative");
+    }
     prunePeriod = period;
   }
 
```

Comment:
Not necessary. I think "period" is the more natural term for describing the property, so it's just a little unfortunate that the best JSE component is named "Duration."